staging/src/k8s.io/apimachinery/pkg/util/net/http.go

```diff
@@ -68,6 +68,9 @@ func IsProbableEOF(err error) bool {
 	if uerr, ok := err.(*url.Error); ok {
 		err = uerr.Err
 	}
+	if neterr, ok := err.(net.Error); ok && neterr.Timeout() {
```

Comment:
...what do you think about logging errors with %#v if we get to the end of this function and are about to return false? I feel like the resulting log noise (which people will complain vociferously about) would help us find the next missing one and/or help us be confident that we finally found them all.

Reply:
I realized there were actually enough mixed callers of this that I split the timeout error detection to a new method and used it specifically in the watch cases. The places where we return errors that aren't handled already complain loudly in the log, e.g.:

```
reflector.go:...: watch of *v1.ConfigMap ended with: an error on the server ("unable to decode an event from the watch stream: net/http: request canceled (Client.Timeout or context cancellation while reading body)") has prevented the request from succeeding
```

Reply:
The main thing that was missing before was not good logging, it was good test coverage of timeout/watch timeout scenarios, running directly and via proxies. This PR adds that coverage.

Reply:
SG, and that also means that the error logging didn't even help us find this :(

Reply:
actually, the existing logging did help (it was what spit out the timeout error and pointed me in the right direction when I saw this issue)